support-lib/jni/djinni_support.cpp

```diff
@@ -239,8 +239,13 @@ jint JniEnum::ordinal(JNIEnv * env, jobject obj) const {
 
 LocalRef<jobject> JniEnum::create(JNIEnv * env, jint value) const {
     LocalRef<jobject> values(env, env->CallStaticObjectMethod(m_clazz.get(), m_staticmethValues));
+    jniExceptionCheck(env);
     DJINNI_ASSERT(values, env);
-    return LocalRef<jobject>(env, env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()), value));
+    LocalRef<jobject> result(env,
+                             env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()),
+                                                        value));
+    jniExceptionCheck(env);
+    return std::move(result);
```

Comment:
The `const` is actually not meaningless because the compiler cannot know whether the variable where RVO stores the result is declared as `const`. If it isn't then creating a `const` variable in that location would result in UB if the caller accessed it as non-`const`. You can still make locals `const` just prefer not to do so on those you intend to return from the function. And remember RVO only kicks in if you return the *exact same* variable in all control paths, otherwise you only get a move.